src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/cas2bail/Cas2BailApplicationEntity.kt

```diff
@@ -0,0 +1,115 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas2bail
+
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.*
+
+
+import io.hypersistence.utils.hibernate.type.json.JsonType
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.LockModeType
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.OneToMany
+import jakarta.persistence.OneToOne
+import jakarta.persistence.Table
+import org.hibernate.annotations.Immutable
+import org.hibernate.annotations.OrderBy
+import org.hibernate.annotations.Type
+import org.springframework.data.domain.Slice
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Lock
+import org.springframework.data.jpa.repository.Query
+import org.springframework.stereotype.Repository
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+import kotlin.jvm.Transient
+
+@Suppress("TooManyFunctions")
+@Repository
+interface Cas2BailApplicationRepository : JpaRepository<Cas2BailApplicationEntity, UUID> {
+  @Query(
+    "SELECT a FROM Cas2BailApplicationEntity a WHERE a.id = :id AND " +
+      "a.submittedAt IS NOT NULL",
+  )
+  fun findSubmittedApplicationById(id: UUID): Cas2BailApplicationEntity?
+
+  @Query("SELECT a FROM Cas2BailApplicationEntity a WHERE a.createdByUser.id = :id")
+  fun findAllByCreatedByUserId(id: UUID): List<Cas2BailApplicationEntity>
+
+  @Query(
+    "SELECT a FROM Cas2BailApplicationEntity a WHERE a.submittedAt IS NOT NULL " +
+      "AND a NOT IN (SELECT application FROM Cas2BailAssessmentEntity)",
+  )
+  fun findAllSubmittedApplicationsWithoutAssessments(): Slice<Cas2BailApplicationEntity>
+}
+
+@Repository
+interface Cas2BailLockableApplicationRepository : JpaRepository<Cas2BailLockableApplicationEntity, UUID> {
+  @Query("SELECT a FROM Cas2BailLockableApplicationEntity a WHERE a.id = :id")
+  @Lock(LockModeType.PESSIMISTIC_WRITE)
+  fun acquirePessimisticLock(id: UUID): Cas2BailLockableApplicationEntity?
+}
+
+@Entity
+@Table(name = "cas_2_bail_applications")
+data class Cas2BailApplicationEntity(
+  @Id
+  val id: UUID,
+
+  val crn: String,
+
+  @ManyToOne
```

Comment:
Can we please make all @ManyToOne relationships Lazy loaded instead of Eager (FetchType.EAGER)